### DISCUSSION TITLE
Release 0.5.1 (1.20.60.24)

### DISCUSSION THREAD

**Discussion by minecraftchloe** [Mojang/minecraft-editor#162] [Announcements]
_[Read the previous release notes here](https://github.com/Mojang/minecraft-editor/discussions/159)_ 

--- 

## Summary 

Hello! Happy Preview day and happy 2024!😊 We are now beginning our iterations towards our next v0.6 release, tentatively titled the “Visualization Update,” where we will be focused on adding more visually engaging elements to the building experience, such as renders of paste previews, new UI components, expanded tools, and brush tip visualizations for more precise placement. 

Read on for more details on this week’s exciting updates! 

As a reminder of Editor API support:  
- Editor Extensions [overview article](https://learn.microsoft.com/en-us/minecraft/creator/documents/editorextensionsintroduction) and  [API reference documentation](https://aka.ms/EditorAPI) on learn.microsoft.com 
- [Starter Kit](https://github.com/Mojang/minecraft-editor-extension-starter-kit): Contains the compiler tools and various libraries required to get started making Editor Extensions 
- [Samples](https://github.com/Mojang/minecraft-editor-extension-samples) created by our team to jump-start your own Extension creations 

Additionally, if you're new to the Editor GitHub this week, we encourage you to join the Bedrock Add-Ons Discord [#Editor channel](https://discord.gg/TZnQKFwGpt)! 

 

---  

## Fixed this release  

- Removed addSeparator function from IMenu API    
- Fixed issue where resource packs were not being retained during editing and testing - now they are retained!  
- Added addEntityPicker function to IPropertyPane API for selecting an entity type   
- Added interface IPropertyItemOptionsDataPicker for both entity and block picker optional properties, removed IPropertyItemOptionsBlocks   
- Introducing Void World generation! The Flat World toggle found on the Advanced tab of the Create New Project page has been replaced with a drop down, allowing new Editor projects to be created in a Vanilla, Flat, or Void world. The Void world project experience will be iterated on in the coming months. Upon starting a Void world project, swap Selection to Fixed Distance to place blocks to kickstart your build. 
- Added toolGroupId optional property to IRegisterExtensionOptionalParameters API for organizing tools in custom groups! 

---  

Thank you for being here! You're the best! ❤️

**Comment by Hommam-Prom** [comment]
Thanks for you
